internal/k8s/reconciler/listener.go

```diff
@@ -51,7 +33,7 @@ type K8sListener struct {
 	listener        gw.Listener
 	client          gatewayclient.Client
 
-	status         ListenerStatus
+	status         *status.ListenerStatus
```

Comment:
For now, just passing a pointer to the status since all of the validation logic is contained here. Pretty much this entire struct is going to go away and we'll just set status values directly on `ListenerState` inside of the validator.